src/env_defaults.h

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2021-2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// These values are copied manually in cunumeric.settings and there is a Python
+// unit test that will maintain that these values and the Python settings
+// values agree. If these values are modified, the corresponding Python values
+// must also be updated.
+
+// 1 << 20 (need actual number for python to parse)
+#define MIN_GPU_CHUNK_DEFAULT 1048576
+#define MIN_GPU_CHUNK_TEST 2
+
+// 1 << 14 (need actual number for python to parse)
+#define MIN_CPU_CHUNK_DEFAULT 16384
+#define MIN_CPU_CHUNK_TEST 2
+
+// 1 << 17 (need actual number for python to parse)
+#define MIN_OMP_CHUNK_DEFAULT 131072
+#define MIN_OMP_CHUNK_TEST 2
+
+#define EAGER_FRACTION_DEFAULT 16
+#define EAGER_FRACTION_TEST 1
+
+#define FAST_MATH_DEFAULT 0
+#define FAST_MATH_TEST 0
```

Comment:
This doesn't currently have a separate value in test-mode

Reply:
Right these are a bit awkward, since `getenv` is used for this env var, not `extract_env`.  I think we can define them for the sake of the Python side as we are doing now, even though they are not especially useful on the cpp side (though I would regard them as "correct"). If you want we can remove them, but we will probably need to make updates back in the legate repo for how they are handled in the docs first, in that case.

Reply:
or we could change `src/cunumeric/cudalibs.cu` to use `extract_env` I assume that's feasible in ` .cu` file

Reply:
I think it's reasonable to have some env-only options that don't have a separate default for testing, e.g. if they're meant to activate a special mode or additional output or something like that.

AFAICT if we just leave `test_default` unset, then the "test mode default" will not be shown at all in the output, which would be slightly cleaner for options that aren't really affected by test-mode. In that case I think all we need to change is `test_settings.py:test_test_default`, to check that `EnvOnlySetting.test_default` is set if and only if `OPTION_TEST` exists in the header (and compare their values of course).

We could also enforce the use of `extract_env` uniformly as you say, that would also be ok, but probably more work.

Reply:
I am just rather frustrated at tedious back-and-forth between multiple repos over things that would be trivial to work on as a unit in a monorepo :man_shrugging: 

My last pitch: FWIW I do think that the notion of "not having a test default" can be legitimately mapped to "the test default is simply identical to the normal default", in a uniform way. (i.e. the statement is literally true: when `LEGATE_TEST=1` the default [therefore, by definition: the *test default*] value is "0")

Regardless, since this `#define` is not actually used I think I'd rather just remove them entirely, and the tests for it here in `cunumeric`. Since it isn't actually coupled to the value in the cpp code in a meaningful way the way the `extract_env` values are, the test does not actually maintain anything.

Reply:
I went ahead and did this in https://github.com/nv-legate/cunumeric/pull/893/commits/e55051e36f3b967e989c3caad817c4f5bd4a76f5 I was going to show how omitting `test_default` currently renders, but suddenly the docs are not building for some entirely unrelated reason
```
/home/bryan/work/cunumeric/cunumeric/random/__init__.py:docstring of cunumeric.random.binomial:54:Footnote [1] is not referenced.
```

Reply:
test default removed in https://github.com/nv-legate/cunumeric/pull/893/commits/479a9e99815afbfa2a19ac03e0c270ac2f1bc586 seems to render OK